packages/nextjs/app/feed/page.tsx

```diff
@@ -0,0 +1,28 @@
+import ContentPage from "./contentPage";
+import FeedHeaderPage from "./header";
+import LeftPage from "./leftPage";
+import RightPage from "./rightPage";
+import type { NextPage } from "next";
+
+const FeedPage: NextPage = () => {
```

Comment:
Might be worth adding a TODO here: If the user is not registered, we redirect them back to `/`

Example: https://github.com/juannyG/coms6998/blob/dev/packages/nextjs/app/home/page.tsx#L29-L38

(Would love your thoughts on if there's a way to add this check without having to put it on _every `page.tsx`_)

Reply:
I have an idea about the Single Sign On (SSO) feature. I could implement some protective functions in a .tsx file that manages the root page. It seems I could modify app/layout.tsx to incorporate redirection logic.https://stackoverflow.com/questions/78319443/how-to-do-azure-single-sign-on-with-next-js-14-and-app-router